### ISSUE TITLE
Using ANNOY as classifier for data classification 

### ISSUE THREAD

**Issue by QueSabz** [spotify/annoy#593]
I have implemented the following annoy class so that I can use it as a classifier but the only issue I have is that it does not have a scoring method. How can I implement scoring method so that this class becomes a fully fledged classifier?

from collections import Counter
#Wrapper for using annoy.AnnoyIndex as sklearn's KNeighborsTransformer
class Annoy():
    
    def __init__(self,n_neighbors=5, metric='euclidean', n_trees=10):
        self.n_neighbors = n_neighbors
        self.metric = metric
        self.n_trees = n_trees
    
    def fit(self, X_train, y_train):
        self.N_feat = X_train.shape[1]
        self.N_train = X_train.shape[0]
        self.y_train = y_train
        self.t = annoy.AnnoyIndex(self.N_feat,metric=self.metric)
        for i, v in zip(range(self.N_train), X_train):
            self.t.add_item(i, v)
        self.t.build(self.n_trees)
        return self

    def predict(self,X_test):
        y_hat = []
        for tv in X_test:
            nn_inds = self.t.get_nns_by_vector(tv, self.n_neighbors)
            nn_classes =[self.y_train[nn] for nn in nn_inds]
            y_hat.append(most_frequent(nn_classes))
        return y_hat 

def most_frequent(List): 
    occurence_count = Counter(List) 
    return occurence_count.most_common(1)[0][0]

**Comment by erikbern** [comment]
You'll have to query for the nearest neighbors, look up their classes, and use that as a score. Basically what https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html does
